lichbot.lic

```diff
@@ -122,10 +123,87 @@ class Lichbot
+  end
+
+  def balance(character)
+    @last_action = Time.now
+    unless has_bot_account(character)
+      fput "whisper #{character} Apologies, but my record shows that you do not have a Bot Bank account."
+      pause 0.5
+      fput "whisper #{character} You can tip me in Kronars to start one."
+      return
+    end
+
+    fput "whisper #{character} Your current balance is: #{@bot_bank[character]} Kronar."
+  end
+
+  def deposit(character, amount, denomination)
+    # TODO: Handle when multiple withdraw/deposit requests are being made (queue?)
+    # TODO: Handle case when player tries to give coins directly to bot (without using TIP). OR, if they
+    #       do not use the TIP command, check against current copper in-hand to ensure you actually received the
+    #       amount given, to avoid spoofing a 'give'
+    @last_action = Time.now
+    if denomination =~ /(Lirum|Dokora)/i
+      fput "whisper #{character} Unfortunately, I'm only accepting Kronars at the moment."
+      bput("decline tip", 'You decline .* offer', 'But you have no tip offers outstanding.')
+      return
+    end
+    
+    has_account = has_bot_account(character)
+
+    case bput("accept tip", "You accept #{character}'s tip and slip it away", "#{character}'s tip offer has expired.")
+    when /.*\'s tip offer has expired./i
+      fput "whisper #{character} Sorry! I was busy and could not accept your deposit in time. Try again."
+    when /You accept .*'s tip and slip it away/i
+      msg = has_account ? "Thank you for your business! Your new balance is: " : "You've officially started a new Bot Account. Your balance is: "
+
+      @bot_bank[character] = has_account ? (get_balance(character) + amount.to_i) : amount.to_i
+      fput "whisper #{character} #{msg} #{@bot_bank[character]} Kronar!"
+    end
+
+    # Save the bot bank to UserVars
+    UserVars.bot_bank = @bot_bank
+  end
+
+  def withdraw(character, amount)
+    # TODO: Handle when multiple withdraw/deposit requests are being made (queue?)
+    @last_action = Time.now
+    unless has_bot_account(character)
+      fput "whisper #{character} Apologies, but my record shows that you do not have a Bot Bank account."
+      pause 0.5
+      fput "whisper #{character} You can tip me in Kronars to start one."
+      return
+    end
+
+    balance = get_balance(character)
+    if balance < amount.to_i
+      fput "whisper #{character} The max you can withdraw is: #{balance} Kronar."
+      return
+    end
+
+    case bput("tip #{character} #{amount} kronar", '\w+ declines your tip offer.', 'Your tip offer to \w+ has expired.', '\w+ accepts your tip and slips it away with a smile')
+    when /\w+ declines your tip offer./i
+    when /Your tip offer to \w+ has expired./i
+      fput "whisper #{character} Alright then, I will continue to hold on to it."
+    when /\w+ accepts your tip and slips it away with a smile/i
+      balance = balance - amount.to_i
```

Comment:
Completely agree on writing to a flat file for both redundancy and as the actual checkbook for this.